inst/private/python_ipc_pytave.m

```diff
@@ -59,17 +61,50 @@
 
   info.prelines = 0;
 
-  % TODO: bit wasteful to repeat this every call
-  headers = python_header();
-  pyexec(headers)
+  if isempty(have_headers)
+    pyexec(strjoin({'import sys',
+                    'import sympy',
+                    'import sympy as sp',
+                    'from sympy import __version__ as spver',
+                    'from sympy import *',
+                    'from sympy.logic.boolalg import Boolean, BooleanFunction',
+                    'from sympy.core.relational import Relational',
+                    '# temporary? for piecewise support',
+                    'from sympy.functions.elementary.piecewise import ExprCondPair',
+                    'from sympy.integrals.risch import NonElementaryIntegral',
+                    'from sympy.matrices.expressions.matexpr import MatrixElement',
+                    '# for hypergeometric',
+                    'from sympy.functions.special.hyper import TupleArg',
+                    'from sympy.utilities.iterables import uniq',
+                    'import copy',
+                    'import binascii',
+                    'import struct',
+                    'import codecs',
+                    'import xml.etree.ElementTree as ET',
+                    'from distutils.version import LooseVersion',
```

Comment:
You should be able to drop `xml`, `binascii`.

Probably/maybe `codecs`, `copy`, `struct` too.

The SymPy stuff is probably all needed: this is because these things appear in `srepr` for some SymPy object, so we need them to copy objects *back* to SymPy.   Worse, they need to be in the main namespace...  Possibly `git blame inst/private/python_header.py` will tell you more about each of them.